packages/core-data/src/actions.js

```diff
@@ -632,6 +657,34 @@ export function* saveEditedEntityRecord( kind, name, recordId, options ) {
 	yield* saveEntityRecord( kind, name, record, options );
 }
 
+/**
+ * Adds all specified entity records to a batch, and then saves them using one or more batch requests.
+ *
+ * @param {Object[]} spec List of { kind, name, key, record } of records to save.
+ * @return {Object} Finalized batch object.
+ */
+export function* __experimentalBatchSaveEntityRecords( spec ) {
+	for ( const { kind, type, key, record } of spec ) {
+		if ( key ) {
```

Comment:
This check seems weird, why sometimes we have a key and other times a record

Reply:
If we're editing an existing record then there is a `key`. If we're creating a new record then there is no `key`.